### PR TITLE
API-945: Connectivity command line test

### PR DESCRIPTION
**Pull request by pierrejolly**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/Connection/CreateConnectionCommandEndToEnd.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+use Akeneo\Test\Integration\Configuration;
+use Symfony\Bundle\FrameworkBundle\Console\Application;
+use Symfony\Component\Console\Tester\CommandTester;
+
+/**
+ * @author Pierre Jolly <pierre.jolly@akeneo.com>
+ * @copyright 2020 Akeneo SAS (http://www.akeneo.com)
+ * @license http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ */
+class CreateConnectionCommandEndToEnd extends WebTestCase
```
Comment (fitn) [inline src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/Connection/CreateConnectionCommandEndToEnd.php]: Why do you extend WebTestCase?
I think you could create your own CommandTestCase and avoid creating twice the kernel :)

File: src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/Connection/CreateConnectionCommandEndToEnd.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+{
+    public function test_it_creates_a_connection(): void
+    {
+        $kernel = static::createKernel();
+        $application = new Application($kernel);
```
Comment (fitn) [inline src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/Connection/CreateConnectionCommandEndToEnd.php]: This should be part of the setUp

File: src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/Connection/CreateConnectionCommandEndToEnd.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+        $command = $application->find('akeneo:connectivity-connection:create');
```
Comment (fitn) [inline src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/Connection/CreateConnectionCommandEndToEnd.php]: This also could be part of the setUp of this test

File: src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/Connection/UpdateAuditDataCommandEndToEnd.php [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+        $stmt = $this->dbalConnection->prepare($sqlQuery);
+        $stmt->execute([
+            'author' => $author,
+        ]);
+    }
+
+    protected function getConfiguration(): Configuration
+    {
+        return $this->catalog->useFunctionalCatalog('catalog_modeling');
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/Connection/UpdateAuditDataCommandEndToEnd.php]: I wonder if creating product in the minimal catalog is not better. @fitn ?

Comment (willy-ahva) [reply src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/Connection/UpdateAuditDataCommandEndToEnd.php]: Of course, it is! Loading a full catalog is far longer than creating your own products or whatever you need in your test.  For example https://github.com/akeneo/pim-community-dev/blob/master/src/Akeneo/Tool/Bundle/ElasticsearchBundle/tests/integration/UpdateIndexMappingIntegration.php#L85

**Pull request closed without merging**